Mod2/Lessons/Week4/LINQ.md

```diff
@@ -0,0 +1,123 @@
+# LINQ
+
+## Learning Goals
+* Understand why we use LINQ
+* Practice basic LINQ methods
+* Recognize LINQ Query Syntax
+
+## Warm Up
+
+```c#
+var names = new List<string>() {
+    "Connor",
+    "Megan",
+    "Charlie",
+    "Molly"
+};
+```
+
+For the `names` list above, use `for` or `foreach` to get the following:
+* A list of names that start with "C"
+* A list of names that are 5 letters long
+* 🌶️The first name that contains an 'e'
+
+## LINQ
+
+When working with collections (lists, arrays, dictionaries), we often need to use `for` or `foreach` to manipulate that list to give us some infomration about it.  We might need to get a subset of the collection, find a specific element in the collection, or get some information _about_ a collection (like if the collection contains a specific object).  These kinds of functions come up so often that many languages create special methods to handle working with collections.
+
+In C#, these additional methods are in the LINQ package.  Most C# developers use LINQ methods to accomplish what you just did in the warm-up.  LINQ makes these tasks easier.
+
+We are going to use [this repo](https://github.com/turingschool-examples/Launch_LinqLesson) to learn some LINQ methods.  Fork and clone the repo, and take about 5 minutes to familiarize yourself with the current code.
+
+### Where
+
+When you are trying to get a **subset** of a collection, you can use the LINQ method `WHERE`.
+
+Compare the two approaches for filtering a list below:
+
+```c#
+var names = new List<string>() {
+    "Connor",
+    "Megan",
+    "Charlie",
+    "Molly"
+};
+
+// Using `foreach`
+var cNamesForeach = new List<string>();
+
+foreach (var name in names)
+{
+    if (name[0] == 'C')
+    {
+        cNamesForeach.Add(name);
+    }
+}
+
+// Using LINQ
+
+var cNamesLinq = names.Where(name => name[0] == 'C');
+```
+
+`Where` is a method that takes a [lambda expression]() as an argument.  In this case, the lambda expression is telling `Where` _how_ to filter the collection.  We want to look at each `name` and filter to only those names where `name[0] == 'C'` is true.
```

Comment:
Are you going to add a link to the Where documentation here?

Reply:
I wish this documentation was a bit easier to understand :/ https://learn.microsoft.com/en-us/dotnet/api/system.linq.queryable.where?view=net-7.0

All the predicate stuff might be more confusing than it's worth, but it still might be helpful to start introducing students to this documentation?

Reply:
oops - I was actually going to link back to the pre-work :)